langlib/lang.value/src/main/ballerina/value.bal

```diff
@@ -177,6 +214,11 @@ public isolated function fromBalString(string s) returns anydata|error = @java:M
 # immutable values.
 # This panics if parameter `v` has cycles.
 #
+# ```ballerina
+# record {|int index; string name;|} student = {index: 12, name: "John"};
```

Comment:
Let's use an open record that allows additional `anydata` fields - `record { int index; string name; }`.

With the current record, a user doesn't have to usually use `toJson`, unless they want to be able to update `index` and `name` to have any `json` value (instead of `int` and `string` only, respectively).

They can just do 

```ballerina
record {|int index; string name;|} student = {index: 12, name: "John"};
json studentJson = student;
```